euphonic/spectra.py

```diff
@@ -1297,6 +1298,122 @@ def from_dict(cls: Type[T], d: Dict[str, Any]) -> T:
                    x_tick_labels=d['x_tick_labels'],
                    metadata=d['metadata'])
 
+def apply_kinematic_constraints(spectrum: Spectrum2D,
+                                e_i: Quantity = None,
+                                e_f: Quantity = None,
+                                angle_range: Tuple[float] = (0, 180.)
+                                ) -> Spectrum2D:
+    """
+    Set events to NaN which violate energy/momentum limits:
+
+      - Energy transfer greater than e_i
+      - q outside region accessible for given e_i and angle range
+
+    This requires x_data to be in wavevector units and y_data to be energy.
+
+    Either e_i or e_f should be set, according to direct/instrument
+    geometry. The other values will be inferred, interpreting y_data as
+    energy transfer.
+
+    Parameters
+    ----------
+    spectrum: input 2-D spectrum, with |q| x_data and energy y_data
+    e_i: incident energy of direct-geometry spectrometer
+    e_f: final energy of indirect-geometry spectrometer
+    angle_range: min and max scattering angles (2θ) of detector bank in degrees
```

Comment:
If no type is specified, the description of each parameter should be indented on the following line, and there should be no `:` https://numpydoc.readthedocs.io/en/latest/format.html#parameters